### PR TITLE
feat(CC): add global connection bandwidth associate resource

### PR DESCRIPTION
**Pull request by profoundwu**
**What this PR does / why we need it**:
add global connection bandwidth associate resource

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add global connection bandwidth associate resource
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST="./huaweicloud/services/acceptance/cc" TESTARGS="-run TestAccGCBAssociate_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cc -v -run TestAccGCBAssociate_basic -timeout 360m -parallel 4
=== RUN   TestAccGCBAssociate_basic
=== PAUSE TestAccGCBAssociate_basic
=== CONT  TestAccGCBAssociate_basic
--- PASS: TestAccGCBAssociate_basic (400.55s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cc        400.604s

```

### CODE REVIEW

File: huaweicloud/services/cc/resource_huaweicloud_cc_global_connection_bandwidth_associate.go [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+	}
+
+	respBody, err := utils.FlattenResponse(resp)
+	if err != nil {
+		return err
+	}
+
+	results := utils.PathSearch("gcbandwidths[?result!='success']", respBody, make([]interface{}, 0)).([]interface{})
+	if len(results) > 0 {
+		messages := utils.PathSearch("[*].message", results, make([]interface{}, 0)).([]interface{})
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/cc/resource_huaweicloud_cc_global_connection_bandwidth_associate.go:124]:
```suggestion
		messages := utils.PathSearch("[*].message", results, []interface{}).([]interface{})
```

File: huaweicloud/services/cc/resource_huaweicloud_cc_global_connection_bandwidth_associate.go [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+	if err != nil {
+		return fmt.Errorf("error disassociating the resource instance from the global connection bandwidth: %s", err)
+	}
+
+	respBody, err := utils.FlattenResponse(resp)
+	if err != nil {
+		return err
+	}
+
+	results := utils.PathSearch("gcbandwidths[?result!='success']", respBody, make([]interface{}, 0)).([]interface{})
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/cc/resource_huaweicloud_cc_global_connection_bandwidth_associate.go:271]:
```suggestion
	results := utils.PathSearch("gcbandwidths[?result!='success']", respBody, []interface{}).([]interface{})
```

File: huaweicloud/services/cc/resource_huaweicloud_cc_global_connection_bandwidth_associate.go [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+					"resource_id":   utils.PathSearch("id", rawMap, nil),
+					"resource_type": utils.PathSearch("type", rawMap, nil),
+					"region_id":     utils.PathSearch("region_id", rawMap, nil),
+					"project_id":    utils.PathSearch("project_id", rawMap, nil),
+				})
+			}
+		}
+	}
+	return result
+}
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/cc/resource_huaweicloud_cc_global_connection_bandwidth_associate.go]:
```suggestion
func flattenGlobalConnectionBandwidthAssociate(resources interface{}) []interface{} {
	if _, ok := resources.([]interface{}); !ok || len(resources.([]interface{})) == 0 {
		return nil
	}
	result := make([]interface{}, 0, len(resources.([]interface{})))
	for _, raw := range resources.([]interface{}) {
		rawMap := raw.(map[string]interface{})
		result = append(result, map[string]interface{}{
			"resource_id":   utils.PathSearch("id", rawMap, nil),
			"resource_type": utils.PathSearch("type", rawMap, nil),
			"region_id":     utils.PathSearch("region_id", rawMap, nil),
			"project_id":    utils.PathSearch("project_id", rawMap, nil),
		})
	}

	return result
}
```

File: huaweicloud/services/acceptance/cc/resource_huaweicloud_cc_global_connection_bandwidth_associate_test.go [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+
+	rc := acceptance.InitResourceCheck(
+		rName,
+		&obj,
+		getGCBAssociateFunc,
+	)
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			acceptance.TestAccPreCheck(t)
+			acceptance.TestAccPreCheckEpsID(t)
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/acceptance/cc/resource_huaweicloud_cc_global_connection_bandwidth_associate_test.go]: looks it's not necessary to use eps ID in this test

File: huaweicloud/services/cc/resource_huaweicloud_cc_global_connection_bandwidth_associate.go (hunk 3 above)
Comment (profoundwu) [reply huaweicloud/services/cc/resource_huaweicloud_cc_global_connection_bandwidth_associate.go]: fixed.

File: huaweicloud/services/acceptance/cc/resource_huaweicloud_cc_global_connection_bandwidth_associate_test.go (hunk 4 above)
Comment (profoundwu) [reply huaweicloud/services/acceptance/cc/resource_huaweicloud_cc_global_connection_bandwidth_associate_test.go]: fixed.

**Pull request merged**